src/mappings/mappingHandlers.ts

```diff
@@ -102,56 +101,76 @@ async function handleValue(vkeys: any[], blockNumber: number) {
       const grp = groups[assetId];
       const len = grp.length;
       const isEven = len % 2 === 0;
-      let dat: { value: string; timestamp: number };
+      let data: { value: string; timestamp: number };
       // handle middle
       if (isEven) {
         const ind = Math.floor(len / 2);
         const it1 = JSON.parse(grp[ind]);
         const it2 = JSON.parse(grp[ind - 1]);
         //
-        dat = {
+        data = {
           value: ((BigInt(it1.value) + BigInt(it2.value)) / BigInt(2)).toString(),
           timestamp: it1.timestamp
         };
       } else {
         const { value, timestamp } = JSON.parse(grp[Math.floor(len / 2)]);
-        dat = {
+        data = {
           value: BigInt(value).toString(),
           timestamp
         };
       }
 
-      if (assetId === relayAssetId) {
-        const lptokens = await getLptokens();
+      const record = AssetValue.create({
+        id: `${blockNumber}-${assetId}`,
+        blockHeight: blockNumber,
+        assetId: Number(assetId),
+        value: data.value,
+        blockTimevalue: Math.floor(data.timestamp / 1000).toString()
+      });
+      logger.info(
+        `dump new asset value at[${blockNumber}] assetId[${assetId}] value[${data.value}]`
+      );
+      await record.save();
+
+      if (relayAssetId === assetId) {
+        // add LP token price
 
-        const lpRecords = lptokens.map(lp => {
+        lpTokens.forEach(async lp => {
           const value = new BigNumber(lp.baseAssetAmount)
-            .times(dat.value)
+            .times(data.value)
             .times(2)
-            .dividedBy(lp.supply);
+            .dividedBy(lp.supply)
+            .toFixed(0);
+
+          logger.info(
+            `dump lpToken asset value at[${blockNumber}] assetId[${assetId}] value[${data.value}]`
+          );
 
-          return AssetValue.create({
+          const record = AssetValue.create({
             id: `${blockNumber}-${lp.id}`,
             blockHeight: blockNumber,
             assetId: Number(lp.id),
-            value: parseInt(value.toString()).toString(),
-            blockTimevalue: Math.floor(dat.timestamp / 1000).toString()
+            value,
+            blockTimevalue: Math.floor(data.timestamp / 1000).toString()
           });
+          await record.save();
```

Comment:
await in forEach may occurs unexpected behavior, use for loop

Reply:
I don't think it's necessary, my asynchronous operations are all independent of each other, I'm guessing you're referring to the uninterruptible nature of asynchrony in `forEach`

Reply:
Great!